build/linux/installer/scripts/dcr-config-parser.rb

```diff
@@ -0,0 +1,52 @@
+#!/usr/local/bin/ruby
+
+require 'fileutils'
+require 'json'
+require_relative 'ConfigParseErrorLogger'
+
+@os_type = ENV['OS_TYPE']
+@controllerType = ENV['CONTROLLER_TYPE']
+@containerType = ENV['CONTAINER_TYPE']
+@dcrConfigFilePattern = '/etc/mdsd.d/config-cache/configchunks/*.json'
+@logs_and_events_streams = %w[
+  CONTAINER_LOG_BLOB
+  CONTAINERINSIGHTS_CONTAINERLOGV2
+  KUBE_EVENTS_BLOB
+  KUBE_POD_INVENTORY_BLOB
+]
+@logs_and_events_only = false
+
+return if !@os_type.nil? && !@os_type.empty? && @os_type.strip.casecmp('windows').zero?
+return unless ENV['USING_AAD_MSI_AUTH'].strip.casecmp('true').zero?
+
+if !@controllerType.nil? && !@controllerType.empty? && @controllerType.strip.casecmp('daemonset').zero? \
+  && @containerType.nil?
+  begin
+    file_path = Dir.glob(@dcrConfigFilePattern).first
```

Comment:
In scenarios like with syslog or private link or customer has arbitery DCRs, first config json file may not  be container insights extension config and this could be possible syslog config (with syslog) and in case of AMPLS scenario, this could be DCE config etc.. we would need enumerate all the files which matching the criterion and get the path for container insights extension config JSON file.